src/AstarteManager.tsx

```diff
@@ -118,7 +118,7 @@ const AstarteProvider = ({
     const authConfig = _.pick(session, ['realm', 'token']);
     const clientConfig = _.merge({}, apiConfig, authConfig);
     return new AstarteClient(clientConfig);
-  }, [config]);
+  }, [config, session]);
```

Comment:
Just a side note here. The `session` dependency implies that we're now instantiating a new AstarteClient each time the session changes. It's not an issue per se, since AstarteClient does not keep an internal state, but we need to check all `useEffect` hooks in pages that use AstarteClient: they have `[astarte.client]` as dependency and will be re-run each time a new astarte.client is instantiated.